bundler/lib/bundler/templates/newgem/github/workflows/build-gems.yml.tt

```diff
@@ -0,0 +1,65 @@
+---
+name: Build gems
+
+on:
+  push:
+    tags:
+      - "v*"
+      - "cross-gem/*"
+  workflow_dispatch:
+
+jobs:
+  ci-data:
+    runs-on: ubuntu-latest
+    outputs:
+      result: ${{ steps.fetch.outputs.result }}
+    steps:
+      - uses: oxidize-rb/actions/fetch-ci-data@v1
+        id: fetch
+        with:
+          supported-ruby-platforms: |
+            exclude: ["arm-linux", "x64-mingw32"]
+          stable-ruby-versions: |
+            exclude: ["head"]
+
+  source-gem:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+
+      - name: Build gem
+        run: bundle exec rake build
+
+      - uses: actions/upload-artifact@v3
```

Comment:
https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/